ext/FluxnetSimulations.jl

```diff
@@ -0,0 +1,308 @@
+        FT,
+        SW_d,
+        LW_d,
+        start_date,
+        θs = zenith_angle,
+        earth_param_set = earth_param_set,
+    )
+
+    MODIS_LAI_path = ClimaLand.Artifacts.get_modis_lai_fluxnet_data(site_ID)
+    MODIS_LAI, header = readdlm(MODIS_LAI_path, ',', header = true)
+    dates = DateTime.(MODIS_LAI[:, 1])
+
+    LAI_dt = Second(dates[2] - dates[1]).value
+    LAI_seconds_since_start_date =
+        [Second(date - start_date).value for date in dates]
+    LAI = TimeVaryingInput(LAI_seconds_since_start_date, FT.(MODIS_LAI[:, 2]))
+    maxLAI = FT(maximum(MODIS_LAI[:, 2]))
+
+    return (; atmos, radiation, LAI, maxLAI)
+end
+
+"""
+    snow_precip_fraction(air_temp, hum)
+
+Estimate the fraction of precipitation that is in snow form,
+given the air temperature at the surface in K and the relative humidity
+(between 0 and 1).
+
+See Jennings, K.S., Winchell, T.S., Livneh, B. et al.
+Spatial variation of the rain–snow temperature threshold across the
+Northern Hemisphere. Nat Commun 9, 1148 (2018).
+https://doi.org/10.1038/s41467-018-03629-7
+"""
+function snow_precip_fraction(air_temp, hum)
+    air_temp_C = air_temp - 273.15
+    α = -10.04
+    β = 1.41
+    γ = 0.09
+    snow_frac = (1.0 / (1.0 + exp(α + β * air_temp_C + γ * hum)))
+    return snow_frac
+end
+
+function set_fluxnet_ic!(
+    Y,
+    site_ID,
+    start_date,
+    hour_offset_from_UTC,
+    model::LandModel,
+)
+    fluxnet_csv_path = ClimaLand.Artifacts.experiment_fluxnet_data_path(site_ID)
+    driver_data = readdlm(fluxnet_csv_path, ',')
+    columns = driver_data[1, :]
+    labels = ("SWC_F_MDS_1", "TS_F_MDS_1", "TA_F")
+    indices = [findfirst(columns .== label) for label in labels]
+    index_map = Dict(zip(labels, indices))
+
+    LOCAL_DATETIME = DateTime.(format.(driver_data[2:end, 1]), "yyyymmddHHMM")
+    UTC_DATETIME = LOCAL_DATETIME .+ Dates.Hour(hour_offset_from_UTC)
+    idx_start = findmin(abs.(UTC_DATETIME .- start_date))
+
+    if index_map["SWC_F_MDS_1"] isa Nothing
+        @. Y.soil.ϑ_l = model.soil.parameters.ν / 2
+    else
+        soil_data = driver_data[2:end, index_map["SWC_F_MDS_1"]]
+        not_missing_mask = .~ismissing.(soil_data)
+        idx_start = argmin(abs.(UTC_DATETIME[not_missing_mask] .- start_date))
+        Y.soil.ϑ_l .= soil_data[not_missing_mask][idx_start] ./ 100
+    end
+    Y.soil.θ_i .= 0
+
+    if index_map["TS_F_MDS_1"] isa Nothing
+        atmos_T_data = driver_data[2:end, index_map["TA_F"]]
+        not_missing_mask = .~ismissing.(atmos_T_data)
+        idx_start = argmin(abs.(UTC_DATETIME[not_missing_mask] .- start_date))
+        T_0 = atmos_T_data[not_missing_mask][idx_start] + 273.15
+    else
+        soil_T_data = driver_data[2:end, index_map["TS_F_MDS_1"]]
+        not_missing_mask = .~ismissing.(soil_T_data)
+        idx_start = argmin(abs.(UTC_DATETIME[not_missing_mask] .- start_date))
+        T_0 = soil_T_data[not_missing_mask][idx_start] + 273.15
```

Comment:
I think a function will help here too, and we can use it for the else and if blocks above too:
```suggestion
function get_var_data(driver_data, varname, utc_datetime, start_date, preprocess_func)
        data = driver_data[2:end, index_map[varname]]
        not_missing_mask = .~ismissing.(data)
        idx_start = argmin(abs.(utc_datetime[not_missing_mask] .- start_date))
        return preprocess_func.(data[not_missing_mask][idx_start])
    end
    celsius_to_kelvin(x) = x + 273.15
    T_0 = get_var_data(driver_data, "TS_F_MDS_1", utc_datetime, start_date, celsius_to_kelvin)
```

Reply:
nice! will add

(Note - TVI for spatial fields does take in a preprocess function, but not for 0D. :( )